### DISCUSSION TITLE
Grateful for Your Contribution & Inquiry About Admin Console

### DISCUSSION THREAD

**Discussion by Frank-linkin** [keycloakify/keycloakify#727] [General]
I would like to express my heartfelt gratitude for your outstanding contribution to the open-source community with Keycloakify. Keycloakify’s clear documentation and powerful features have saved us considerable time, and we are truly thankful for your work.
On another note, I noticed that there’s mention of future plans to develop the Admin Console. I am very excited about this feature! I was wondering if there’s any update on its progress or a tentative release timeline?
Once again, thank you and your team for your incredible contributions to the community

**Comment by garronej** [comment]
Hello @Frank-linkin,

First of all, thank you for taking the time to express your appreciation for the project. I truly appreciate it!

Regarding the Admin UI support, I'm actively working on it. I could have shipped it already, but I realized the development experience could be improved by implementing an alternative approach. 

The current Account SPA customization support in Keycloakify has some flaws. For instance, it's difficult to determine which files need to be ejected, and once that's figured out, the process is still manual and quite inconvenient. Additionally, a major issue is that, after ejecting a component, only the ejected parts use the new version, while the non-ejected parts of the Account UI still rely on the old version.

To solve this, I've implemented a new system that will be introduced with the Account UI customization support, but it still needs some refinement. I also need to test whether it's a practical development experience for people who don't work on high-end MacBook. If not, I'll need to find a way to enable true hot module replacement. Waiting more than 20 seconds to see live changes is not acceptable.

I'll keep you posted once it's ready. I expect it to be released in the next couple of months, but please don't hold me to that — it could take more or less time depending on the blockers I encounter along the way, as this work is quite experimental.

On a side note, I don’t usually ask, but the user base that might benefit from Admin UI customization is relatively small — most teams only need a login theme. If this feature is important to your business, you might consider supporting the project via GitHub Sponsors to help make the undertaking more sustainable. This would allow me to justify spending time on it instead of focusing solely on the Login UI.

Of course, there's no obligation to donate. Keycloakify will remain free and open source, but any contribution would be greatly appreciated.

Best,

**Reply by Frank-linkin** [reply]
Good idea. thanks for your efforts.

**Reply by garronej** [reply]
Thank you **very** much @Frank-linkin for your donation! It's much appreciated!

**Comment by garronej** [comment]
Hello @Frank-linkin,

The Admin Console customization support is finally complete!
And it’s not just a “better than nothing” solution; I genuinely think it offers a solid developer experience and should be relatively easy to maintain moving forward.

I’m currently in the process of rewriting the entire documentation, which might take a couple of weeks to finish. However, if you’d like to start using it early, don’t hesitate to reach out. We can schedule a quick session on Discord to get you up and running.  

Best,
